Makefile

```diff
@@ -0,0 +1,55 @@
+# Etherpad-lite Docker Makefile
+COMPOSE     = docker-compose
+COMPOSE_RUN = $(COMPOSE) run --rm
+WAIT_DB     = $(COMPOSE_RUN) dockerize -wait tcp://postgresql:5432 -timeout 60s
+
+default: help
+
+# -- Project
+private/SESSIONKEY.txt:
+	mkdir -p private
+	head /dev/urandom | tr -dc A-Za-z0-9 | head -c 32 > private/SESSIONKEY.txt
+
+private/APIKEY.txt:
+	mkdir -p private
+	head /dev/urandom | tr -dc A-Za-z0-9 | head -c 32 > private/APIKEY.txt
+
+# -- Docker/compose
+build: ## build the etherpad container
+	@$(COMPOSE) build etherpad
+.PHONY: build
+
+down: ## remove stack (warning: it removes the database container)
+	@$(COMPOSE) down
+.PHONY: down
+
+logs: ## display etherpad logs (follow mode)
+	@$(COMPOSE) logs -f etherpad
+.PHONY: logs
+
+run: \
+  private/SESSIONKEY.txt \
+  private/APIKEY.txt
+run: ## start the server
+	@$(COMPOSE) up -d postgresql
+	@$(WAIT_DB)
+	@$(COMPOSE) up -d etherpad
```

Comment:
Instead of doing that in the Makefile, you can do the following directly in the `docker-compose.yml` : 

```
  etherpad:
    [...]
    depends_on:
      - "dockerize"

  dockerize:
    image: jwilder/dockerize
    depends_on:
      - "postgresql"
    command: dockerize -wait "tcp://postgresql:5432" -timeout 60s
```


And then, in your Makefile, you just have to run `$(COMPOSE) up -d etherpad`. WDYT ?

Reply:
Good idea! It's more elegant to avoid being Makefile-dependant for such feature. :+1:

Reply:
In fact, this is not working because compose does not wait for the `dockerize` command to finish before starting dependencies... every service starts and `etherpad`'s bootstrap fails when the database connection could not be established.

Reply:
Oh, sorry I was too much confident on this one :confused:

Reply:
I was also enthusiastic on this! :wink: